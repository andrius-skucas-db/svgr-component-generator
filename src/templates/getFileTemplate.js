const generateImports = files => {
  return files.reduce((accumulator, file) => {
    return `${accumulator}
import ${file.componentName} from './${file.name}';`;
  }, ``);
};

const generateMap = files => {
  return files.reduce((accumulator, file) => {
    return `${accumulator}${accumulator ? "," : ""}
  ['${file.componentName}', ${file.componentName}]`;
  }, ``);
};

const generatePropTypes = files => {
  return files.reduce((accumulator, file) => {
    return `${accumulator}${accumulator ? "," : ""}
    '${file.componentName}'`;
  }, ``);
};

const getFileTemplate = (componentName, filesToInclude) => {
  return `
import React from 'react';
import PropTypes from 'prop-types';    
${generateImports(filesToInclude)}
    
const iconsMap = new Map([${generateMap(filesToInclude)}
]);
    
const ${componentName} = ({ icon, ...rest}) => {
  const Icon = iconsMap.get(icon);

  return Icon ? <Icon {...rest} /> : null;
};

${componentName}.propTypes = {
  icon: PropTypes.oneOf([${generatePropTypes(filesToInclude)}
  ])
};

export default ${componentName};
  `;
};

export default getFileTemplate;
