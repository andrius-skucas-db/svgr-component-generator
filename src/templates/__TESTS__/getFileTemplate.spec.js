import getFileTemplate from "../getFileTemplate";

describe("getFileTemplate", () => {
  it("getFileTemplate", () => {
    const filesToInclude = [
      {
        name: "icon.svg",
        componentName: "Icon"
      },
      {
        name: "icon-2.svg",
        componentName: "Icon2"
      }
    ];
    const testResult = getFileTemplate("IconComponent", filesToInclude);

    expect(testResult).toMatchSnapshot();
  });
});
