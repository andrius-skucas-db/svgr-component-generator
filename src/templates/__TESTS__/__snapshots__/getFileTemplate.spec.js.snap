// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getFileTemplate getFileTemplate 1`] = `
"
import React from 'react';
import PropTypes from 'prop-types';    

import Icon from './icon.svg';
import Icon2 from './icon-2.svg';
    
const iconsMap = new Map([
  ['Icon', Icon]
  ['Icon2', Icon2]
]);
    
const IconComponent = ({ icon, ...rest}) => {
  const Icon = iconsMap.get(icon);

  return Icon ? <Icon {...rest} /> : null;
};

IconComponent.propTypes = {
  icon: PropTypes.oneOf([,
    'Icon',
    'Icon2'])
};

export default IconComponent;
  "
`;
